### DISCUSSION TITLE
Run JiT inside the browser

### DISCUSSION THREAD

**Discussion by LoganDark** [tailwindlabs/tailwindcss#4087] [Ideas]
l really like to use Tailwind because Tailwind is great for rapid prototyping. You never leave the HTML markup when you're designing your element, that's powerful. No more back and forth between the HTML markup and the CSS styles. And no more spending 30 seconds looking up `@media` queries when you can just slap a `md:` or `sm:` down instead.

l also really like rapid prototyping using my [HTML editor](https://html.logandark.net/). No external build system, no project setup, just one file and your code.

l feel like this is a perfect use case for the JiT compiler. ln a development environment were l CBA to spend 30 minutes wrangling Webpack and PostCSS and Tailwind configs, l would love to be able to slap a `<script>` tag down and have a JiT compiler generate all the styles l need. Namely because the CDN version of Tailwind is _extremely_ limited, and projects like [Ridiculous Tailwind](https://github.com/logandark/ridiculous-tailwind) are very large and unwieldy. (not to mention out of date)

So, what about a "CDN JiT compiler" that l can include via a `<script>` tag to generate Tailwind classes on the fly, in-browser?

**Comment by mcmire** [comment]
There may be a way to do this using web workers, but my guess is that it would be very difficult considering that a lot of the heavy lifting behind Tailwind is being done by PostCSS, which operates in Node-land. Tailwind would either have to ship with a tiny server that an in-browser piece could talk to or would have to reimplement the whole kit and caboodle on the client side.
